dcplib/etl/__init__.py

```diff
@@ -73,28 +72,36 @@ def extract_transform_one(self, bundle_uuid, bundle_version, transformer: callab
 
             return tb
 
+    def page_bundles(self, query=None, replica="aws", page_size=None):
+        if query is None:
+            yield from self.dss_client.get_bundles_all.paginate(replica=replica, per_page=page_size)
+        else:
+            yield from self.dss_client.post_search.paginate(es_query=query, replica=replica, per_page=page_size)
+
     def extract(self, query=None, max_workers=512, max_dispatchers=1, page_size=500,
                 dispatch_executor_class: concurrent.futures.Executor = concurrent.futures.ThreadPoolExecutor,
                 transformer: callable = None, loader: callable = None, finalizer: callable = None,
                 page_processor: callable = None):
         start = time.time()
         if query is None:
-            query = self.default_bundle_query
-        total_bundles = self.dss_client.post_search(es_query=query, replica="aws")["total_hits"]
-        if total_bundles == 0:
-            logger.error("No bundles found, nothing to do")
-            return
+            total_bundles = 1
+        else:
+            total_bundles = self.dss_client.post_search(es_query=query, replica="aws")["total_hits"]
+            if total_bundles == 0:
+                logger.error("No bundles found, nothing to do")
+                return
         logger.info("Scanning %s bundles", total_bundles)
         extracted_bundle_count, error_bundle_count = 0, 0
         with dispatch_executor_class(max_workers=max_workers) as executor:
-            for page in self.dss_client.post_search.paginate(es_query=query, replica="aws", per_page=page_size):
+            for page in self.page_bundles(query=query, replica="aws", page_size=page_size):
                 logger.info(f"Extracted bundles: {extracted_bundle_count} "
                             f"({extracted_bundle_count/total_bundles:.1%}, {error_bundle_count} errors)")
```

Comment:
Nit - I think this will be misleading when query is None/total bundles is hard coded to 1

Reply:
Agreed - still looking for a solution for this one. Ideally we'd have a way to find the total # of bundles before paging through each one. Although since we're caching files to disk, perhaps the most reasonable thing to do is to save all bundle IDs to disk first, count them, and read them back. It will rob us of a tiny bit of parallelism (we can't start ETLing individual bundles until we're done writing the list of all bundles) but I expect that effect to be negligible.

Reply:
👍